src/language/pattern-types.md

```diff
@@ -59,6 +204,39 @@ case [a, b]: // ...
 case const [a, b]: // ...
 ```
 
+## Variable
+
+`var bar, String str, final int _`
+
+Variable patterns bind new variables to values that have been matched or destructured. 
+They usually occur as part of a [destructuring pattern][destructure] in order to
+capture a destructured value.
+
+The variables are in scope in a region of code that is only reachable when the
+pattern has matched.
+
+<?code-excerpt "language/lib/patterns/pattern_types.dart (variable)"?>
+```dart
+switch ((1, 2)) {
+  // 'var a' and 'var b' are variable patterns that bind to 1 and 2, respectively.
+  case (var a, var b): // ...
+  // 'a' and 'b' are in scope in the case body.
+}
+```
+
+A _typed_ variable pattern only matches if the matched value has the declared type,
+and fails otherwise:
```

Comment:
**question:** So this pattern match matches on type annotation, not variable value?

Reply:
Yup, it's a pattern that declares new variables. So the value going into the declaration only matters in the sense of "can this value bind to this new variable". So in the case of a "regular" variable pattern, the value doesn't really matter because like with any variable declaration you can just write `var a = ` followed by pretty much anything (major generalization, but sufficient for this topic). But, as is the case for any variable declaration, if you annotate the declaration like `int a = `, then the value needs to adhere to the annotation.